packages/core/src/config.ts

```diff
@@ -360,6 +362,25 @@ export function composeBannerFooterConfig(
   };
 }
 
+export function composeDecoratorsConfig(
+  options: ts.CompilerOptions,
+  version?: NonNullable<
+    NonNullable<RsbuildConfig['source']>['decorators']
+  >['version'],
+): RsbuildConfig {
+  if (version || !options.experimentalDecorators) {
+    return {};
+  }
+
+  return {
+    source: {
+      decorators: {
+        version: 'legacy',
```

Comment:
why defaults to legacy

Reply:
This is the return value of `1&&2`
1. no `source.decorators.version` config
2. `experimentalDecorators`: true in tsconfig

Reply:
Maybe I should write on the contrary

```ts
  if (!version && options.experimentalDecorators) {
    return {
    source: {
      decorators: {
        version: 'legacy',
      },
    },
  };
  }

  return {};
```

Reply:
My mistake, the current implementation is ok